### DISCUSSION TITLE
Are Getters reactive by default?

### DISCUSSION THREAD

**Discussion by bobmcallan** [vuejs/pinia#513] [Help and Questions] [answered]
Vue 3, using setup;

`        setup() {
            const user = userStore();

            return {
                username: computed(() => user.username),
            };
        }
`

or 

`        setup() {
            const user = userStore();

            return {
                username: user.getUsername,
            };
        }
`

Which is reactive, and if getters are not, then why have them?

**Comment by posva** [comment, marked as answer]
Getters are computed properties under the hood. I don't really understand your question. Another way of writing it is:

```js
setup() {
  const user = useUserStore()
  const username = toRef(user, 'username')
  // or if extracting multiple ones
  const { username } = toRefs(user)
  return {
    username
  }
}
```

**Comment by bobmcallan** [comment]
Thanks, you highlighted my question. "Getters are computed properties under the hood.", 

However in the setup code I need to wrap with "username: computed(() => user.username),". Does this not double up computed(computed(variable))? Or am I missing something?

Bob

**Reply by posva** [reply]
It doesn't double up. You can also use toRef(store, 'username') instead of creating a computed. You can also return the whole store in setup.
